openstack/clientconfig/results.go

```diff
@@ -126,6 +126,20 @@ type AuthInfo struct {
 	// DefaultDomain is the domain ID to fall back on if no other domain has
 	// been specified and a domain is required for scope.
 	DefaultDomain string `yaml:"default_domain,omitempty" json:"default_domain,omitempty"`
+
+	// AllowReauth should be set to true if you grant permission for Gophercloud to
+	// cache your credentials in memory, and to allow Gophercloud to attempt to
+	// re-authenticate automatically if/when your token expires.  If you set it to
+	// false, it will not cache these settings, but re-authentication will not be
+	// possible.  This setting defaults to false.
+	//
+	// NOTE: The reauth function will try to re-authenticate endlessly if left
+	// unchecked. The way to limit the number of attempts is to provide a custom
+	// HTTP client to the provider client and provide a transport that implements
+	// the RoundTripper interface and stores the number of failed retries. For an
+	// example of this, see here:
+	// https://github.com/rackspace/rack/blob/1.0.0/auth/clients.go#L311
+	AllowReauth *bool `yaml:"allow_reauth,omitempty" json:"allow_reauth,omitempty"`
```

Comment:
Why is this `*bool` instead of `bool`? If it was `bool`, I believe you could remove your code in `openstack/clientconfig/requests.go` on line 376-382.

Reply:
It's because I wanted to set default value for it in the code. 

I think your point is that `bool` default value is `false` and therefore it would basically set this value to its default (`AllowReauth` should be `false` by default). Do I understand correctly?

I can do it like that, it's just general approach - if the default was `true` I couldn't set it without potentially overwriting user's value.

I am not that experienced with Golang, so I will do as you prefer.

Reply:
I updated to `bool` -- also to avoid complex initialization in multiple places (nil pointers threat)